sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1952,4 +1952,49 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
       }
     }
   }
+
+  test("create datasource table with an non-existent location") {
+    withTable("t", "t1") {
+      withTempDir {
+        dir =>
+          dir.delete()
```

Comment:
there are lots of dir existed test case in DDLSuit or HiveDDLSuit, so we just add non-existent test cases